sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientJavaDocCodeSamples.java

```diff
@@ -0,0 +1,140 @@
+            }, error -> System.err.print(error.toString()));
+        // END: com.azure.messaging.servicebus.servicebusasyncreceiverclient.receive#message
+
+        // When program ends, or you're done receiving all messages.
+        subscription.dispose();
+    }
+
+    /**
+     * Receives message with back pressure.
+     */
+    public void receiveBackpressure() {
+
+        ServiceBusReceiverAsyncClient receiver = new ServiceBusClientBuilder()
+            .connectionString("fake-string")
+            .buildReceiverClientBuilder()
+            .queueName("<QUEUE-NAME>")
+            .buildAsyncClient();
+
+        // BEGIN: com.azure.messaging.servicebus.servicebusasyncreceiverclient.receive#basesubscriber
+        receiver.receive().subscribe(new BaseSubscriber<ServiceBusReceivedMessage>() {
+            private static final int NUMBER_OF_MESSAGES = 5;
+            private final AtomicInteger currentNumberOfMessages = new AtomicInteger();
+
+            @Override
+            protected void hookOnSubscribe(Subscription subscription) {
+                // Tell the Publisher we only want 5 message at a time.
+                request(NUMBER_OF_MESSAGES);
+            }
+
+            @Override
+            protected void hookOnNext(ServiceBusReceivedMessage value) {
+                // Process the ServiceBusReceivedMessage
+
+                // If the number of messages we have currently received is a multiple of 5, that means we have reached
+                // the last message the Subscriber will provide to us. Invoking request(long) here, tells the Publisher
+                // that the subscriber is ready to get more messages from upstream.
+                if (currentNumberOfMessages.incrementAndGet() % 5 == 0) {
+                    request(NUMBER_OF_MESSAGES);
+                }
+            }
+        });
+        // END: com.azure.messaging.servicebus.servicebusasyncreceiverclient.receive#basesubscriber
+    }
+
+    /**
+     * Receives from all the messages.
+     */
+    public void receiveAll() {
+        ServiceBusReceiverAsyncClient receiver = new ServiceBusClientBuilder()
+            .connectionString("fake-string")
+            .buildReceiverClientBuilder()
+            .buildAsyncClient();
+
+        // BEGIN: com.azure.messaging.servicebus.servicebusasyncreceiverclient.receive#all
+
+        Disposable subscription = receiver.receive().subscribe(receivedMessage -> {
+            String messageId = receivedMessage.getMessageId();
+
+            System.out.printf("Received message messageId %s%n", messageId);
+            System.out.printf("Contents of message as string: %s%n", new String(receivedMessage.getBody(), UTF_8));
+        });
+
+        // When program ends, or you're done receiving all messages.
+        subscription.dispose();
+        // END: com.azure.messaging.servicebus.servicebusasyncreceiverclient.receive#all
+    }
```

Comment:
You need to dispose of consumer as well, like you did in other samples. (even if it doesn't appear in the snippet.)